### PR TITLE
fix scikit-image bug #1680 (grayscale png round trip fails) and add u…

### PR DESCRIPTION
**Pull request by connellybarnes**
Fix bug #1680

### CODE REVIEW

File: skimage/io/tests/test_pil.py [hunk 1]
```diff
@@ -39,6 +39,13 @@ def setup_module(self):
     except ImportError:
         pass
 
+def test_png_round_trip():
+    f = NamedTemporaryFile(suffix='.png')
+    fname = f.name
```
Comment (blink1073) [inline skimage/io/tests/test_pil.py]: I think you need to close the file handle here.

Comment (connellybarnes) [inline skimage/io/tests/test_pil.py]: I don't think so. When f is garbage collected at the end of the function, the temporary file will be removed per the NamedTemporaryFile spec: "If delete is true (the default), the file is deleted as soon as it is closed."

Comment (blink1073) [inline skimage/io/tests/test_pil.py]: I was referring to the test failure on Appveyor, which was due to a permissions error on this file. 



—

**Comment by blink1073** [comment]
Looks great, thanks @connellybarnes, let's wait for Travis-CI to agree.

**Comment by blink1073** [comment]
Thanks!

**Pull request merged**